src/RA/ResponseContext.cs

```
@@ -220,54 +220,81 @@ private void Initialize()
             ParseLoad();
         }
 
-        private void Parse()
+        /// <summary>
+        /// Holds response parser functions
+        /// </summary>
+        private static Dictionary<string, Func<string, dynamic>> parsers = new Dictionary<string, Func<string, dynamic>>()
         {
-            var contentType = ContentType();
-
-            if (contentType.Contains("json"))
-            {
-                if (!string.IsNullOrEmpty(_content))
+            { "json",  delegate(string instr)
                 {
                     try
                     {
-                        _parsedContent = JObject.Parse(_content);
-                        return;
+                        return JObject.Parse(instr);
                     }
                     catch
                     {
+                        try
+                        {
+                            return JArray.Parse(instr);
+                        }
+                        catch(Exception e)
+                        {
+                            throw e;

```

Please note that this is breaking change. Original code silently swallows exception (which is not nice indeed), but your change throws. I would agree with your change, however it would be worth mentioning it in readme alongside feature you have added.

Also want to check, way you catch and throw - is that intentional? Do we even need `try-catch` block there if we want exception to be thrown anyway?

---

I don't think this is a breaking change. While it originally swallowed the exception when parsing, if it got to the end without managing to parse it would throw an exception if the _content wasn't empty. Mine checks if it's an empty context at the beginning rather than the end, then throws the exception from that step instead of at the end.